e2e_tests/src/e2e_conduit/fixtures/importers/follower_algod.py

```diff
@@ -0,0 +1,115 @@
+    def __init__(self, sourcenet):
+        self.algoddir = None
+        self.sourcenet = sourcenet
+        self.last = None
+        super().__init__()
+
+    @property
+    def name(self):
+        return "algod_follower"
+
+    @property
+    def lastblock(self):
+        if self.last is None:
+            raise RuntimeError("algod importer has no blockfiles configured")
+        return self.last
+
+    def resolve_config_input(self):
+        with open(os.path.join(self.algoddir, "algod.net"), "r") as algod_net:
+            self.config_input["netaddr"] = "http://" + algod_net.read().strip()
+        with open(os.path.join(self.algoddir, "algod.token"), "r") as algod_token:
+            self.config_input["token"] = algod_token.read().strip()
+
+    def resolve_config_output(self):
+        self.config_output["algod_token"] = self.config_input["token"]
+        self.config_output["algod_net"] = self.config_input["netaddr"]
+        self.config_output["algod_data_dir"] = self.algoddir
+
+    def setup(self, accumulated_config):
+        try:
+            conduit_dir = accumulated_config["conduit_dir"]
+        except KeyError as exc:
+            logger.error(
+                f"AlgodImporter needs to be provided with the proper config: {exc}"
+            )
+            raise
+        source_is_tar = hassuffix(
+            self.sourcenet, ".tar", ".tar.gz", ".tar.bz2", ".tar.xz"
+        )
+        if not (source_is_tar or (self.sourcenet and os.path.isdir(self.sourcenet))):
+            # Assuming self.sourcenet is an s3 source net
+            tarname = f"{self.sourcenet}.tar.bz2"
+            # fetch test data from S3
+            bucket = "algorand-testdata"
+
+            s3 = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+            prefix = "indexer/e2e4"
+            if "/" in tarname:
+                cmhash_tarnme = tarname.split("/")
+                cmhash = cmhash_tarnme[0]
+                tarname =cmhash_tarnme[1]
+                prefix+="/"+cmhash
+                tarpath = os.path.join(conduit_dir, tarname)
+            else:
+                tarpath = os.path.join(conduit_dir, tarname)
+            success = firstFromS3Prefix(s3, bucket, prefix, tarname, outpath=tarpath)
+            if not success:
+                raise Exception(
+                    f"failed to locate tarname={tarname} from AWS S3 path {bucket}/{prefix}"
+                )
+            source_is_tar = True
+            self.sourcenet = tarpath
+        tempnet = os.path.join(conduit_dir, "net")
+        if source_is_tar:
+            xrun(["tar", "-C", conduit_dir, "-x", "-f", self.sourcenet])
+        else:
+            xrun(["rsync", "-a", self.sourcenet + "/", tempnet + "/"])
+        blockfiles = glob.glob(
+            os.path.join(conduit_dir, "net", "Primary", "*", "*.block.sqlite")
+        )
+        self.last = countblocks(blockfiles[0])
+        for root, dirs, files in os.walk(os.path.join(tempnet, 'Node', 'tbd-v1')):
```

Comment:
Very clever. A comment here might have helped, I jumped to the wrong conclusion when I saw you fetching the dataset.
```suggestion
        # Reset the secondary node, and enable follow mode.
        # This is what conduit will connect to for data access.
        for root, dirs, files in os.walk(os.path.join(tempnet, 'Node', 'tbd-v1')):
```